File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap{};
        for (int i = 0; i < nums.size(); i++) {
            auto it = umap.find(target - nums.at(i));
            if (it != umap.end())
                return vector<int> {it->second, i};
            
            umap[nums.at(i)] = i;
        }

        return vector<int>{};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int l = 0, r = numbers.Length - 1;

        while (l < r) {
            int sum = numbers[l] + numbers[r];

            if (sum == target) return [l + 1, r + 1];

            if (sum < target) {
                l++;
            } else {
                r--;
            }
        }

        return [-1, -1];
    }
}
